Orange/widgets/visualize/owvenndiagram.py

```diff
@@ -531,27 +530,24 @@ def extract_rowwise(self, var_dict, ids=None, selection=False):
                         else:
                             values[atr_type].append(v[perm].reshape(-1, 1))
                 else:
-                    var = var_data[1][0]
-                    value = np.empty((len(all_ids), 1))
-                    value.fill(np.nan)
-                    domain[atr_type].append(deepcopy(var_data[1][0][0]))
+                    value = np.full((len(all_ids), 1), np.nan)
+                    domain[atr_type].append(var_data[1][0][0].copy())
                     for _, table_key in var_data[1]:
                         #different tables have different part of the same attribute vector
                         perm = permutations[table_key]
                         v = getattr(self.data[table_key].table[list(ids[table_key].values()),
                                                                var_name],
                                     self.atr_vals[atr_type]).reshape(-1, 1)
-                        if v.dtype != value.dtype:
-                            value = value.astype(v.dtype, copy=False)
+                        value = value.astype(v.dtype, copy=False)
```

Comment:
I believe that it's safe to call `astype(..., copy=False)`: it will return the original array if the type already matches.